serotiny/models/vae/base_vae.py

```diff
@@ -199,38 +197,35 @@ def make_results_dict(
         z_composed,
         x_hat,
     ):
+
         results = {
             "loss": loss,
-            f"{stage}_loss": loss.detach().cpu(),  # for epoch end logging purposes
-            "kld_loss": kld_loss.detach().cpu(),
+            f"{stage}_loss": loss,  # for epoch end logging purposes
+            "kld_loss": kld_loss,
         }
 
         for part, z_comp_part in z_composed.items():
             results.update(
-                {
-                    f"z_composed/{part}": z_comp_part.detach().cpu(),
-                }
+                {f"z_composed/{part}": z_comp_part,}
             )
 
         for part, recon_part in reconstruction_loss.items():
             results.update(
-                {
-                    f"reconstruction_loss/{part}": recon_part.detach().cpu(),
-                }
+                {f"reconstruction_loss/{part}": recon_part,}
             )
 
         for part, z_part in z_parts.items():
             results.update(
                 {
-                    f"z_parts/{part}": z_part.detach().cpu(),
-                    f"z_parts_params/{part}": z_parts_params[part].detach().cpu(),
-                    f"kld/{part}": kld_per_part[part].detach().float().cpu(),
+                    f"z_parts/{part}": z_part,
+                    f"z_parts_params/{part}": z_parts_params[part],
+                    f"kld/{part}": kld_per_part[part],
                 }
             )
 
         if self.hparams.id_label is not None:
             if self.hparams.id_label in batch:
-                ids = batch[self.hparams.id_label].detach().cpu()
+                ids = batch[self.hparams.id_label]
                 results.update({self.hparams.id_label: ids, "id": ids})
 
         return results
```

Comment:
For all of these guys, do we really need to remove the `.detach()` bit? Is removing `.cpu()` not enough? As I understand it, not detaching the values that get logged causes unnecessary memory blow-up because there are gradients that don't get garbage collected. (This might have been fixed since I last checked)